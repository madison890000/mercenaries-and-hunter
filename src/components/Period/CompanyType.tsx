import Tag from '../Tag';
import React from 'react';
import DataModel from '../../models/types';

interface CompanyTypeProps {
    companyType: DataModel.CompanyType;
}

const CompanyType = ({ companyType }: CompanyTypeProps) => (
    <span>
        {companyType === DataModel.CompanyType.Enterprise && <Tag type="less">Enterprise</Tag>}
        {companyType === DataModel.CompanyType.Startup && <Tag type="less">Startup</Tag>}
    </span>
);

export default CompanyType;
