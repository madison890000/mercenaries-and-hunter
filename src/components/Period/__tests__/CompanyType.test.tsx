import React from 'react';
import CompanyType from '../CompanyType';
import renderer from 'react-test-renderer';
import DataModel from '../../../Model/types';

test('renders CompanyType is Enterprise', () => {
    const CompanyTypeDom = renderer.create(
        <CompanyType companyType={DataModel.CompanyType.Enterprise} />
    )
        .toJSON();
    expect(CompanyTypeDom)
        .toMatchSnapshot();
});

test('renders CompanyType is Startup', () => {
    const CompanyTypeDom = renderer.create(
        <CompanyType companyType={DataModel.CompanyType.Startup} />
    )
        .toJSON();
    expect(CompanyTypeDom)
        .toMatchSnapshot();
});
