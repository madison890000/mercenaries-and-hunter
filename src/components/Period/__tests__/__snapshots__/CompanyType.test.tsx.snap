// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CompanyType is Enterprise 1`] = `
<span>
  <span
    className="less"
  >
    Enterprise
  </span>
</span>
`;

exports[`renders CompanyType is Startup 1`] = `
<span>
  <span
    className="less"
  >
    Startup
  </span>
</span>
`;
